### ISSUE TITLE
Minio Helm chart uses default "/bitnami/minio/data" path even when "persistence.mountPath" is provided

### ISSUE THREAD

**Issue by Mo0rBy** [bitnami/charts#21757]
### Name and Version

bitnami/minio 12.8.18 > 12.9.1

### What architecture are you using?

My k8s cluster uses amd64, but I tested this locally on arm64 arch, so it is the same for both.

### What steps will reproduce the bug?

1. Create a PV and PVC for Minio
2. Deploy Minio Helm chart to a cluster (I have used all versions between 12.8.18 > 12.9.1 (inclusively)) with a value defined for `persistence.mountPath` that is not the default `bitnami/minio/data`.
3. Describe the `deployment` object shows to see that the volume is successfully mounted.
4. Shell into the Minio container and `ls` the `/` directory. You can see both the default `bitnami/minio/data` AND your custom volume mount.
5. Create a bucket in the Minio UI and maybe upload a file or 2.
6. Shell into the Minio container and see that the data is still written to `bitnami/minio/data` instead of the your custom `persistance.mountPath`.

### Are you using any custom parameters or values?

Yes, the `persistence.mountPath` value:
```yaml
values:
  persistence:
    existingClaim: minio
    size: 50Gi
    mountPath: /minio
```

### What is the expected behavior?

I would expect the `bitnami/minio/data` directory to not exist OR it does exist and it is not used.
Any data uploaded to Minio should be stored inside the custom `mountPath` directory.

### What do you see instead?

Data is written to the default `bitnami/minio/data` directory and is therefore ephemeral and lost of the the Helm deployment is updated or upgraded.

### Additional information

I'm not sure where access policies are stored but it would be useful if these were not ephemeral as well.
Basically, I need all my data to be stored on my custom PV that has storage provisioned on AWS so that when the deployment is updated, I don't lose all my data.

**Comment by Mo0rBy** [comment]
I've done some more research and came across [this thread](https://github.com/bitnami/containers/issues/48235). Is fixing my issue just a matter of adding an environment variable to my MinIO deployment to change the data dir of the container? If yes, would it be possible to make this change within the Helm chart as well? Then no one else will experience this problem.

**Comment by Mo0rBy** [comment]
I've just tried adding this to my Helm chart deployment:
```
    extraEnvVars:
      - name: MINIO_DATA_DIR
        value: /minio
```

And I get this error: 
```
 10:27:47.39 INFO  ==> ** Starting MinIO setup **
minio 10:27:47.40 DEBUG ==> Validating settings in MINIO_* env vars..
minio 10:27:47.47 INFO  ==> Starting MinIO in background...
ERROR Unable to use the drive /bitnami/minio/data: drive access denied: Invalid arguments specified
minio 10:27:57.50 INFO  ==> Adding local Minio host to 'mc' configuration...
```

So even with this envVar set, Minio is still trying to use the default data dir I think.

**Comment by Mo0rBy** [comment]
> I've just tried adding this to my Helm chart deployment:
> 
> ```
>     extraEnvVars:
>       - name: MINIO_DATA_DIR
>         value: /minio
> ```
> 
> And I get this error:
> 
> ```
>  10:27:47.39 INFO  ==> ** Starting MinIO setup **
> minio 10:27:47.40 DEBUG ==> Validating settings in MINIO_* env vars..
> minio 10:27:47.47 INFO  ==> Starting MinIO in background...
> ERROR Unable to use the drive /bitnami/minio/data: drive access denied: Invalid arguments specified
> minio 10:27:57.50 INFO  ==> Adding local Minio host to 'mc' configuration...
> ```
> 
> So even with this envVar set, Minio is still trying to use the default data dir I think.

Nevermind, I removed the `persistence.mountPath` and didn't realise, using both `persistence.mountPath` and the `MINIO_DATA_DIRz envVar allow the container to start correctly, however, data is still stored in the default `/bitnami/minio/data` location.

**Comment by Mo0rBy** [comment]
I can confirm the `MINIO_DATA_DIR` envVar is being used by looking at my yaml and shelling into the container and executing `env | grep data`:
```
I have no name!@minio-2-7959bf54dc-rrltg:/minio/data$ env | grep data
PWD=/minio/data
MINIO_DATA_DIR=/minio/data
```
But creating a test bucket still create data in the default directory:
```
I have no name!@minio-2-7959bf54dc-rrltg:/minio/data$ cd /bitnami/minio/data/
I have no name!@minio-2-7959bf54dc-rrltg:/bitnami/minio/data$ ls
test-bucket
```

**Comment by Mo0rBy** [comment]
I've also tried using the `containerSecurityContext` options to add an `fsGroup` definition and `runAsUser` definition but data is still stored in the default `/bitnami/minio/data` directory.

**Comment by Mo0rBy** [comment]
Anyone seen this?
As I say, this is either a major bug or I'm missing something very critical.

**Comment by FraPazGal** [comment]
Hello @Mo0rBy, apologies for the radio silence on holidays season! I've been able to reproduce this issue and it seems there is an issue with the `MINIO_DATA_DIR` envar, as it is not been used in the initial minio server setup. If we look for the root cause inside the minio pod we'll see the following:

```
I have no name!@minio-8f4765b56-qw56h:/$ printenv | grep MINIO_DATA_DIR
MINIO_DATA_DIR=/tmp/minio
I have no name!@minio-8f4765b56-qw56h:/$ ps aux
USER       PID %CPU %MEM    VSZ   RSS TTY      STAT START   TIME COMMAND
1001         1  0.2  1.3 1590428 163208 ?      Ssl  16:53   0:01 /opt/bitnami/minio/bin/minio server --certs-dir /certs --console-address :9001 --address :9000 /bitnami/minio/data

```

The server is started via the [run.sh](https://github.com/bitnami/containers/blob/main/bitnami/minio/2023/debian-11/rootfs/opt/bitnami/scripts/minio/run.sh#L21-L36) script. The issue is that the `$MINIO_DATA_DIR` in that script is loaded from:

```
# Load MinIO environment
. /opt/bitnami/scripts/minio-env.sh
```

If we check that env file, we'll see that `MINIO_DATA_DIR` has a fixed value and can't be changed:

```
I have no name!@minio-8f4765b56-qw56h:/$ cat /opt/bitnami/scripts/minio-env.sh | grep MINIO_DATA_DIR
export MINIO_DATA_DIR="/bitnami/minio/data"
```

Let me check with the dev team on the reason this configuration is in place and the best approach to solve this.

**Comment by Mo0rBy** [comment]
No worries @FraPazGal , I thought it would be a while until there was any reply as I did post this on Boxing Day I think and I wasn't expecting anyone to be working. And I'm super happy that it seems to be a bug and not me being stupid.

**Comment by Mo0rBy** [comment]
Am I allowed to raise a PR for this? I can't find any docs on how to raise changes on the README or anything so if you I am allowed and you can point me to the docs, that would much appreciated.

**Comment by Mo0rBy** [comment]
@FraPazGal ^?

**Comment by jerryguowei** [comment]
I am facing the same issues and have to use the default mount path `/bitnami/minio/data`, which is not a problem for me. But  it would be better fix it as the chart supply the `persistence.mountPath`.

**Comment by Mo0rBy** [comment]
@FraPazGal Is there any progress on this? Have the Devs replied with a reason for making the dataDir envVar non-configurable within the container image configs?

**Comment by Mo0rBy** [comment]
Thanks @FraPazGal , I understand this is an issue with the container script, and I will be looking at that contribution guide to see if I can make the fix.

The reason I mention the helm chart is that I believe the `persistence.mountPath` should also pass it's value to the container shell script so that a mounted volume is used as the `DATA_DIR` instead of the default value. Is there a different contribution guide for the Helm chart? Or is it the same? I can look into making this change as well after getting some confirmation that this is indeed what this chart parameter is intended for.

**Comment by FraPazGal** [comment]
Oh, I think I didn't express myself correctly. I agree with you we will also need to modify the chart's templates. It's just that modifying them prior to the container changes won't fix anything. If I'm not mistaken, we'll have to modify both the stateful and deployment templates so they set the `MINIO_DATA_DIR` into the pod. Going by `MINIO_CERTS_DIR` implementation, it could be something like this:

```
          env:
            {{- if .Values.persistence.mountPath }}
            - name: MINIO_DATA_DIR
              value: {{ .Values.persistence.mountPath | quote }}
            {{- end }
```

As for your question, the [chart's contributing guidelines](https://github.com/bitnami/charts/blob/main/CONTRIBUTING.md) are pretty similar to the ones at our container's repo with a few differences. Feel free to ask here if you encounter any issue  and we'll gladly help.

**Comment by Mo0rBy** [comment]
@FraPazGal I noticed in your example you use the `MINIO_CERTS_DIR`, but looking at the same [minio-env.sh](https://github.com/bitnami/containers/blob/2e3623f3e86a036479e0fe78ee95b486bfac1653/bitnami/minio/2024/debian-11/rootfs/opt/bitnami/scripts/minio-env.sh#L59) file, the certs dir is not configurable either LOL.

I can create another PR for this as well if it's something that you want.

**Comment by FraPazGal** [comment]
Damn, good catch @Mo0rBy! If you want you can include the changes for `MINIO_CERTS_DIR` in the same PR! Thanks again for you contributions!

**Comment by Mo0rBy** [comment]
@FraPazGal I have not made the same changes to the certs dir as I havent really tested that change and I'm not aware of where that path is read in other places, so I just left that the same.

**Comment by FraPazGal** [comment]
That's fine @Mo0rBy, I'll note it down and create an internal task to review its behaviour. Let us review the PR and continue the discussion there.

**Comment by kapilgarg1996** [comment]
Hi @FraPazGal , After the update, the old data is not accessible anymore because it resides in /data volume. How can we copy the old keys to new location without losing it ?

**Comment by sass1997** [comment]
@FraPazGal @Mo0rBy: Thank you guys for fixing this issue. What do you think about updating the helm chart to make sure `MINIO_DATA_DIR ` has the values of `{{ .Values.persistence.path }}`?

**Comment by Mo0rBy** [comment]
@sass1997 I am going to start working in these changes this weekend, so watch this space, the changes will be coming soon. The PR linked above for the `bitnami/containers` repo was required first.

**Comment by Mo0rBy** [comment]
I think this issues can now be closed with that merged PR. Thanks for all your help @FraPazGal, I hope to make more submissions in the future.
